### PR TITLE
Add local worker

### PR DESCRIPTION
**Pull request by bakura10**
This PR addresses #19 and adds a local worker.

I was a bit unsure about how it should be tested considering that it's in an infinite loop. I've just tested it manually in different circumstances and it works like a charm, so as it's just a development tool, I think we'll leave it this way except if you have some idea on how to test it properly :).

ping @danizord

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -128,3 +187,108 @@ Beanstalk to retry it later, you should map SilentFailingListener to the message
...
+alongside your `index.php` file:
+
+```php
+require_once 'vendor/autoload.php';
+
+use Symfony\Component\Console\Application;
+use Zend\ServiceManager\ServiceManager;
+use ZfrEbWorker\ModuleConfig;
+
+$config = (new ModuleConfig())->__invoke();
```
Comment (danizord) [inline README.md]: Won't work because it needs application config as well (in order to build the AwsSdk)

File: src/Cli/PublisherCommand.php [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+                'name',
+                null,
+                InputOption::VALUE_REQUIRED,
+                'Message name (eg.: user.created)'
+            )
+            ->addOption(
+                'queue',
+                null,
+                InputOption::VALUE_REQUIRED,
+                'Queue name to pull messages from'
```
Comment (danizord) [inline src/Cli/PublisherCommand.php]: to publish

File: src/Cli/WorkerCommand.php [hunk 3]
```diff
@@ -0,0 +1,164 @@
...
+     * @return void
+     */
+    private function processMessage(
+        array $message,
+        string $server,
+        string $queueName,
+        string $queueUrl,
+        OutputInterface $output
+    ) {
+        $response = $this->httpClient->post(rtrim($server, '/') . '/internal/worker', [
```
Comment (danizord) [inline src/Cli/WorkerCommand.php]: Why don't you let the user specify the full URL via argument? WorkerMiddleware works in any URL.

File: src/Cli/WorkerCommand.php [hunk 4]
```diff
@@ -0,0 +1,164 @@
...
+                'X-Aws-Sqsd-Msgid'             => $message['MessageId'],
+                'X-Aws-Sqsd-Queue'             => $queueName,
+                'X-Aws-Sqsd-First-Received-At' => $message['Attributes']['ApproximateFirstReceiveTimestamp'],
+                'X-Aws-Sqsd-Receive-Count'     => $message['Attributes']['ApproximateReceiveCount'],
+                'X-Aws-Sqsd-Sender-Id'         => $message['Attributes']['SenderId']
+            ],
+            'json' => json_decode($message['Body'], true)
+        ]);
+
+        if ($response->getStatusCode() === 200) {
```
Comment (danizord) [inline src/Cli/WorkerCommand.php:152]: `>= 200 && < 300`?

File: test/Cli/PublisherCommandTest.php [hunk 5]
```diff
@@ -0,0 +1,107 @@
...
+    public function testThrowExceptionIfQueueDoesNotExist()
+    {
+        $input  = $this->prophesize(InputInterface::class);
+        $output = $this->prophesize(OutputInterface::class);
+
+        $input->getOption('name')->shouldBeCalled()->willReturn('user.created');
+        $input->getOption('queue')->shouldBeCalled()->willReturn('default');
+
+        $this->sqsClient->getQueueUrl(['QueueName' => 'default'])->shouldBeCalled()->willThrow(SqsException::class);
+
```
Comment (danizord) [inline test/Cli/PublisherCommandTest.php:71]: Missing `$output->writeln()` expectation.

File: test/Cli/PublisherCommandTest.php [hunk 6]
```diff
@@ -0,0 +1,107 @@
...
+        $input->getOption('name')->shouldBeCalled()->willReturn('user.created');
+        $input->getOption('queue')->shouldBeCalled()->willReturn('default');
+        $input->getOption('payload')->shouldBeCalled()->willReturn('key=value&user[first_name]=John&user[last_name]=Doe');
+
+        $this->sqsClient->getQueueUrl(['QueueName' => 'default'])->shouldBeCalled()->willReturn([
+            'QueueUrl' => 'https://sqs.amazonaws.com'
+        ]);
+
+        $this->queuePublisher->setQueue('default', 'https://sqs.amazonaws.com');
```
Comment (danizord) [inline test/Cli/PublisherCommandTest.php]: Missing `shouldBeCalled()`

File: test/Cli/PublisherCommandTest.php [hunk 7]
```diff
@@ -0,0 +1,107 @@
...
+        $this->queuePublisher->push('default', 'user.created', [
+            'key'  => 'value',
+            'user' => [
+                'first_name' => 'John',
+                'last_name'  => 'Doe'
+            ]
+        ]);
```
Comment (danizord) [inline test/Cli/PublisherCommandTest.php]: Missing `shouldBeCalled()`

File: test/Cli/PublisherCommandTest.php [hunk 8]
```diff
@@ -0,0 +1,107 @@
...
+        $this->queuePublisher->flush();
```
Comment (danizord) [inline test/Cli/PublisherCommandTest.php]: Missing `shouldBeCalled()`

File: test/Cli/WorkerCommandTest.php [hunk 9]
```diff
@@ -0,0 +1,80 @@
...
+    public function testThrowExceptionIfQueueDoesNotExist()
+    {
+        $input  = $this->prophesize(InputInterface::class);
+        $output = $this->prophesize(OutputInterface::class);
+
+        $input->getOption('server')->shouldBeCalled()->willReturn('http://localhost');
+        $input->getOption('queue')->shouldBeCalled()->willReturn('default');
+
+        $this->sqsClient->getQueueUrl(['QueueName' => 'default'])->shouldBeCalled()->willThrow(SqsException::class);
+
```
Comment (danizord) [inline test/Cli/WorkerCommandTest.php:70]: Missing `$output->writeln()` expectation.

File: src/Cli/WorkerCommand.php (hunk 3 above)
Comment (bakura10) [inline src/Cli/WorkerCommand.php]: It's just a sane default so you don't need to type the "/internal/worker" all the time. But I can add a "--path"? that owuld be appended in place of "/internal/worker"?

File: src/Cli/WorkerCommand.php (hunk 4 above)
Comment (bakura10) [inline src/Cli/WorkerCommand.php:152]: From the Beanstalk documentation:

> If the application returns any response other than 200 OK, then Elastic Beanstalk waits to put the message back in the queue after the configured VisibilityTimeout period.

**Comment by bakura10** [comment]
Done @danizord

File: src/Cli/WorkerCommand.php (hunk 3 above)
Comment (danizord) [inline src/Cli/WorkerCommand.php]: Yep, looks good

File: src/Cli/WorkerCommand.php (hunk 4 above)
Comment (danizord) [inline src/Cli/WorkerCommand.php:152]: oh, my bad then :D

**Comment by danizord** [comment]
Looks good for me :+1:

**Pull request merged**